src/Strategy/Go/GoListPackages.hs

```diff
@@ -0,0 +1,240 @@
+--   Stdlib packages with pseudo versions are not filtered yet. This is the same behavior as the current dynamic tactic in Strategy.Go.GoList.
+-- * Removes path dependencies and child deps that aren't used elsewhere in the graph.
+--   The go tools give us this data but we haven't decided yet how to present it.
+-- * Replaces modules according to the module's 'replacement' field.
+buildGraph :: (Has Diagnostics sig m) => [GoPackage] -> m (Graphing.Graphing Dependency, GraphBreadth)
+buildGraph rawPackages =
+  fmap ((,Complete) . pruneUnreachable) -- Remove deps that are only children of removed path deps.
+    . evalGrapher
+    . traverse_ graphEdges
+    $ pkgsNoStdLibImports
+  where
+    (stdLibImportPaths, pkgsNoStdLibImports) = foldl' go (HashSet.empty, HashMap.empty) rawPackages
+
+    go :: (HashSet.HashSet ImportPath, HashMap.HashMap ImportPath GoPackage) -> GoPackage -> (HashSet.HashSet ImportPath, HashMap.HashMap ImportPath GoPackage)
+    go (stdLibPaths, otherPackages) g@GoPackage{standard, importPath}
+      | standard = (HashSet.insert importPath stdLibPaths, otherPackages)
+      | otherwise = (stdLibPaths, HashMap.insert importPath (removeIgnoredPackages g) otherPackages)
+
+    -- "C" is a special package for using Go's FFI.
+    -- Ignore it because trying to look it up as a package import path later will fail.
+    ignoredPackages :: HashSet.HashSet ImportPath
+    ignoredPackages = HashSet.insert (ImportPath "C") stdLibImportPaths
+
+    removeIgnoredPackages :: GoPackage -> GoPackage
+    removeIgnoredPackages g@GoPackage{packageDeps = pDeps} =
+      g{packageDeps = filter (\i -> not $ i `HashSet.member` ignoredPackages) pDeps}
+
+    graphEdges :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => GoPackage -> m ()
+    graphEdges GoPackage{importPath, packageDeps, listError} = do
+      maybe (pure ()) warn listError
+      currModule@GoModule{isMainModule, indirect} <- importToModule importPath
+      unless (isMainModule || isPathDep currModule) $ do
+        let currDep :: Dependency
+            currDep = modToDep currModule
+
+            addChildEdge :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => Dependency -> m ()
+            addChildEdge childDep = do
+              when (childDep /= currDep) $
+                edge currDep childDep
+
+            makeEdge :: (Has (Grapher Dependency) sig m, Has Diagnostics sig m) => ImportPath -> m ()
+            makeEdge = importToModule >=> pure . modToDep >=> addChildEdge
+        traverse_ makeEdge packageDeps
+        if indirect
+          then deep currDep
+          else direct currDep
+
+    -- Look up module info for an import path, performing replacements if necessary.
+    importToModule :: Has Diagnostics sig m => ImportPath -> m GoModule
+    importToModule impPath = do
+      -- Return a stand-in value for missing import paths so it's visible to users? even if unscannable.
```

Comment:
Why the `?`? Is this a leftover TODO wondering what to do if an importPath doesn't have a match in the `HashMap`?

Have you seen this in the example projects you have worked with?

Reply:
You're right, this is a holdover from an earlier version of the code. The only case I've seen this happen is with the special `C` package and that's been filtered out earlier in the process now. If we can't look one up, that means there's something weird with the `go list` output and it outputs a fatal error in that case. I'm going to remove the comment.